pkg/utils/dataset.go

```diff
@@ -89,3 +92,94 @@ func IsTargetPathUnderFluidNativeMounts(targetPath string, dataset datav1alpha1.
 
 	return false
 }
+
+// GetUpdateUFSMap get a map of mount points to be added and removed
+func GetUpdateUFSMap(dataset *datav1alpha1.Dataset) (map[string][]string, error) {
+	updateUFSMap := make(map[string][]string)
+	// 1. get spec MountPaths and mounted MountPaths from dataset
+	specMountPaths, mountedMountPaths, err := getMounts(dataset)
+	if err != nil {
+		return updateUFSMap, err
+	}
+
+	// 2. get mount points which are needed to be added/removed
+	added, removed := calculateMountPointsChanges(specMountPaths, mountedMountPaths)
+	if len(added) > 0 {
+		updateUFSMap["added"] = added
+	}
+
+	if len(removed) > 0 {
+		updateUFSMap["removed"] = removed
+	}
+
+	return updateUFSMap, err
+}
+
+// getMounts get the spec and current mounted mountPaths of dataset
+// No need for a mount point with Fluid native scheme('local://' and 'pvc://') to be mounted
+func getMounts(dataset *datav1alpha1.Dataset) (specMountPaths, mountedMountPaths []string, err error) {
+	for _, mount := range dataset.Spec.Mounts {
+		if common.IsFluidNativeScheme(mount.MountPoint) {
+			continue
+		}
+		alluxioPathInCtx := UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)
+		specMountPaths = append(specMountPaths, alluxioPathInCtx)
+	}
+	for _, mount := range dataset.Status.Mounts {
+		if common.IsFluidNativeScheme(mount.MountPoint) {
+			continue
+		}
+		alluxioPathHaveMounted := UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Status.Mounts)
+		mountedMountPaths = append(mountedMountPaths, alluxioPathHaveMounted)
+	}
+
+	return specMountPaths, mountedMountPaths, err
+}
+
+// calculateMountPointsChanges compare the spec and status of a dataset to calculate MountPoints Changes
+func calculateMountPointsChanges(specMountPaths, mountedMountPaths []string) (add, remove []string) {
+	add = funk.SubtractString(specMountPaths, mountedMountPaths)
```

Comment:
Can we use `SubtractString` rather than importing funk package?

Reply:
I wonder why we use a []sting instead of a set(or map).  Seems it's a diff operation on set data structure